client/wfprev-war/src/main/angular/src/app/components/edit-project/project-details/project-files/project-files.component.ts

```diff
@@ -120,31 +123,55 @@ export class ProjectFilesComponent implements OnInit {
   loadActivityAttachments(): void {
     if (!this.fiscalGuid || !this.activityGuid) return;
     this.projectGuid = this.route.snapshot?.queryParamMap?.get('projectGuid') ?? '';
-    // Find projectPlanFiscalGuid from the activityGuid
-    this.attachmentService.getActivityAttachments(this.projectGuid, this.fiscalGuid, this.activityGuid).subscribe({
-      next: (response) => {
-        if (response?._embedded?.fileAttachment && Array.isArray(response._embedded.fileAttachment)) {
-          const fileAttachments = response._embedded.fileAttachment.sort((a: FileAttachment, b: FileAttachment) => {
-            const timeA = new Date(a.uploadedByTimestamp ?? 0).getTime();
-            const timeB = new Date(b.uploadedByTimestamp ?? 0).getTime();
-            return timeB - timeA; // latest first
+  
+    // Fetch activity boundaries
+    this.projectService.getActivityBoundaries(this.projectGuid, this.fiscalGuid, this.activityGuid).subscribe({
+      next: (boundaryResponse) => {
+        const boundaries = boundaryResponse?._embedded?.activityBoundary;
+  
+        // Build map of activityBoundaryGuid → boundarySizeHa to set polygon hecatares for each file
+        const boundarySizeMap = new Map<string, number>();
+        if (boundaries && boundaries.length > 0) {
+          boundaries.forEach((boundary: { activityBoundaryGuid: string, boundarySizeHa: number }) => {
+            boundarySizeMap.set(boundary.activityBoundaryGuid, boundary.boundarySizeHa);
           });
-          this.projectFiles = fileAttachments;
-          this.dataSource.data = [...this.projectFiles];
-        } else {
-          console.error('Expected an array of activity files inside _embedded.fileAttachment, but got:', response);
-          this.projectFiles = [];
         }
-      },
-      error: (err) => {
-        this.snackbarService.open('Failed to load activity attachments.', 'Close', {
-          duration: 5000,
-          panelClass: 'snackbar-error',
+  
+        this.attachmentService.getActivityAttachments(this.projectGuid, this.fiscalGuid, this.activityGuid).subscribe({
+          next: (response) => {
+            if (response?._embedded?.fileAttachment && Array.isArray(response._embedded.fileAttachment)) {
+              const fileAttachments = response._embedded.fileAttachment.sort((a: FileAttachment, b: FileAttachment) => {
+                const timeA = new Date(a.uploadedByTimestamp ?? 0).getTime();
+                const timeB = new Date(b.uploadedByTimestamp ?? 0).getTime();
+                return timeB - timeA; // latest first
+              });
+  
+              // Set polygonHectares for each attachment
+              this.projectFiles = fileAttachments.map((file: FileAttachment) => ({
```

Comment:
I find we are calling fileAttachments.map((file: FileAttachment) twice, might be worth to create a helper function to lower the duplication. just a suggestion but dont have to be adjust in this pr.